app/controllers/users/emails_controller.rb

```diff
@@ -22,6 +22,10 @@ def add
       result = @add_user_email_form.submit(current_user, permitted_params)
       analytics.add_email_request(**result.to_h)
 
+      if pending_completions_consent?
+        user_session[:pending_completions_consent] = true
+      end
+
```

Comment:
Can you explain what this code is for?

Reply:
It sets a session value that will help determine the redirect in the email confirmations controller https://github.com/18F/identity-idp/blob/f940e20f206fd850dd2071705627dc9a497c36c5/app/controllers/users/email_confirmations_controller.rb#L47-L51

it will be deleted in the completions controller when they get redirected finally to the SP
https://github.com/18F/identity-idp/blob/f940e20f206fd850dd2071705627dc9a497c36c5/app/controllers/sign_up/completions_controller.rb#L31